### PR TITLE
Modify rule S2583: LaYC format

### PR DESCRIPTION
**Pull request by martin-strecker-sonarsource**
[S2583](https://sonarsource.github.io/rspec/#/rspec/S2583/csharp)

## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S2583/metadata.json [hunk 1]
```diff
@@ -34,5 +34,5 @@
   "defaultQualityProfiles": [
     "Sonar way"
   ],
-  "quickfix": "unknown"
+  "quickfix": "partial"
```
Comment (martin-strecker-sonarsource) [inline rules/S2583/metadata.json:37]: The `d ?? "value"` case can be fixed.

File: rules/S2583/csharp/rule.adoc [hunk 2]
```diff
@@ -54,27 +77,10 @@ public void Sample(bool b)
...
-----
-
-* When the condition is the literal ``++true++`` or ``++false++``.
-
-In these cases it is obvious the code is as intended.
-
 include::../see.adoc[]
+* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators#conditional-logical-and-operator-[Microsoft Learn] - Conditional logical AND operator &&
+* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators#conditional-logical-or-operator-[Microsoft Learn] - Conditional logical OR operator ||
+* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator[Microsoft Learn] - ?? and ??= operators - the null-coalescing operators
```
Comment (sebastien-marichal) [inline rules/S2583/csharp/rule.adoc:83]:
```suggestion
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators#conditional-logical-and-operator-[Microsoft Learn - Conditional logical AND operator &&]
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/boolean-logical-operators#conditional-logical-or-operator-[Microsoft Learn - Conditional logical OR operator ||]
* https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator[Microsoft Learn - ?? and ??= operators - the null-coalescing operators]
```

File: rules/S2583/description.adoc [hunk 3]
```diff
@@ -1 +1 @@
-Conditional expressions which are always ``++true++`` or ``++false++`` can lead to dead code. Such code is always buggy and should never be used in production.
+Conditional expressions which are always `true` or `false` can lead to dead code. Such code is always buggy and should never be used in production.
```
Comment (sebastien-marichal) [inline rules/S2583/description.adoc:1]: I would explain what "dead code" is; the user might not be familiar with this term. Also, we could add more details about its impact on an application
A link to the [wiki page](https://en.wikipedia.org/wiki/Dead_code) could be added.
Or maybe, we could change the term to [unreachable code](https://en.wikipedia.org/wiki/Unreachable_code), which is more precise for this rule (based on Wikipedia's definition).

Comment (sebastien-marichal) [inline rules/S2583/description.adoc:1]: I don't agree with this statement or the term `buggy` is not appropriate here.

```suggestion
Conditional expressions which are always `true` or `false` can lead to dead code. Such code should never be used in production.
```

File: rules/S2583/metadata.json (hunk 1 above)
Comment (sebastien-marichal) [inline rules/S2583/metadata.json:37]: The quickfix update should be in `csharp/metadata.json`, not sure we can take this decision for all squads.

**Review by sebastien-marichal — CHANGES_REQUESTED** [review]

File: rules/S2583/description.adoc (hunk 3 above)
Comment (martin-strecker-sonarsource) [reply rules/S2583/description.adoc]: I agree. I changed it now and I'm asking myself what "Such code should never be used in production." is conveying as information to the user? I would say this sentence needs clarification (what is production code) and is also confusing: Why is it okay to do it in non-production code, like test code? What about removing that part as well?

Comment (sebastien-marichal) [reply rules/S2583/description.adoc]: We should remove it, I agree, it does not make sense.
We could replace this part with a small paragraph about the impact with an example such as unreachable code with a `Dispose` call or some other critical statement.

**Review by sebastien-marichal — CHANGES_REQUESTED** [review]
Only left is to add more details in the description, IMO.

**Pull request merged**